operations/mimir/read-write-deployment.libsonnet

```diff
@@ -0,0 +1,452 @@
+    mimir_backend_data_disk_class: 'fast-dont-retain',
+    mimir_backend_allow_multiple_replicas_on_same_node: false,
+
+    // Query-scheduler ring-based service discovery is always enabled in the Mimir read-write deployment mode.
+    query_scheduler_service_discovery_mode: if $._config.read_write_deployment_enabled then 'ring' else super.query_scheduler_service_discovery_mode,
+
+    // Overrides-exporter is part of the backend component in the Mimir read-write deployment mode.
+    overrides_exporter_enabled: if $._config.read_write_deployment_enabled then false else super.overrides_exporter_enabled,
+  },
+
+  local container = $.core.v1.container,
+  local deployment = $.apps.v1.deployment,
+  local envVar = $.core.v1.envVar,
+  local service = $.core.v1.service,
+  local podAntiAffinity = deployment.mixin.spec.template.spec.affinity.podAntiAffinity,
+  local podDisruptionBudget = $.policy.v1beta1.podDisruptionBudget,
+  local pvc = $.core.v1.persistentVolumeClaim,
+  local statefulSet = $.apps.v1.statefulSet,
+  local volumeMount = $.core.v1.volumeMount,
+
+  // Mimir read-write deployment mode makes some strong assumptions about what must enabled enabled and disabled.
+  check_compactor_max_concurrency: if !$._config.read_write_deployment_enabled || $._config.compactor_max_concurrency == 1 then null else
+    error 'please set compactor_max_concurrency to 1 when using Mimir read-write deployment mode',
+
+  check_ingester_multi_zone: if !$._config.read_write_deployment_enabled || $._config.multi_zone_ingester_enabled then null else
+    error 'please set multi_zone_ingester_enabled to true when using Mimir read-write deployment mode',
+
+  check_store_gateway_multi_zone: if !$._config.read_write_deployment_enabled || $._config.multi_zone_store_gateway_enabled then null else
+    error 'please set multi_zone_store_gateway_enabled to true when using Mimir read-write deployment mode',
+
+  check_querier_autoscaling: if !$._config.read_write_deployment_enabled || !$._config.autoscaling_querier_enabled then null else
+    error 'please set autoscaling_querier_enabled to false when using Mimir read-write deployment mode',
+
+  check_ruler_remote_evaluation_enabled: if !$._config.read_write_deployment_enabled || !$._config.ruler_remote_evaluation_enabled then null else
+    error 'please set ruler_remote_evaluation_enabled to false when using Mimir read-write deployment mode',
+
+  check_overrides_exporter_enabled: if !$._config.read_write_deployment_enabled || !$._config.overrides_exporter_enabled then null else
+    error 'please set overrides_exporter_enabled to false when using Mimir read-write deployment mode',
+
+  check_memberlist_ring: if !$._config.read_write_deployment_enabled || $._config.memberlist_ring_enabled then null else
+    error 'please set memberlist_ring_enabled to true when using Mimir read-write deployment mode',
+
+  // Uninstall resources used by microservices deployment mode.
```

Comment:
I guess some day we'll need to either include one or another, instead of uninstalling them.

Reply:
Right. Is this significantly different enough that it warrants a whole new jsonnet project? eg `operations/mimir-read-write-deployment`? Especially since switching is a destructive operation.

Reply:
I don't want separate jsonnet projects. Two projects means doing changes in 2 places every time and it's harder to keep consistency between the two. However I agree with Oleg that a follow up work should be creating the microservices exported resources (deployments, statefulsets, ...) conditionally instead of uninstalling them. I will add as a TODO to my work.